### DISCUSSION TITLE
Log in Ring Account - Problem

### DISCUSSION THREAD

**Discussion by Ricox1975** [koush/scrypted#695] [Q&A] [answered]
i can't login in my ring accout
<img width="1493" alt="Bildschirmfoto 2023-04-05 um 19 50 43" src="https://user-images.githubusercontent.com/52467622/230163208-2b6591dd-ed54-42d0-b104-04bafb069c72.png">

**Comment by imcl75** [comment]
Same issue.  Scrypted plugin for Ring not triggering Ring to send 2FA code.  I also tried with authenticator to generate 2fa code - also did not login.  Scrypted up to date.  This was also occurring with ring homebridge plugin but that is now working again.

**Reply by kimfar** [reply]
Is your node.js up to date?

**Reply by cloudfiretim** [reply]
I have the same issue. I'm running with Docker on a Synology DS720+. Node.js on the Synology is 18.12.0-0005. I see an option to install v16 as well but is that necessary? Where does Scrypted pull node.js from? I know I had to install a separate version for Homebridge that I also have running on the Synology.

**Reply by imcl75** [reply]
Ubuntu linux node.js v 18.15.0 
do i need to check specific node.js for scripted? if so how (sorry)?

**Comment by jordenc** [comment]
Same problem on my Synology, seems they fixed it at https://github.com/dgreif/ring/issues/1192 but I think this is not implemented in the Ring plugin yet.

**Comment by koush** [comment]
https://github.com/koush/scrypted/commit/2dd4721b7f35218c6d998c738d65f847aab6d748

**Comment by koush** [comment, marked as answer]
https://github.com/koush/scrypted/commit/638c1f77fde9affa28f2c6b58085490ceef32db8

**Reply by hjuhlin** [reply]
Looks like I still have the same problem in 0.0.109 as in 0.0.108 :( 

unhandledRejection Error: Refresh token is not valid.  Unable to authenticate with Ring servers.  See https://github.com/dgreif/ring/wiki/Refresh-Tokens
    at xt.getGrantData (C:\external\ring-client-api\packages\ring-client-api\rest-client.ts:186:11)

**Comment by koush** [comment]
clear your user and password and log in again.

**Reply by imcl75** [reply]
Plug in fails to load

linux x64 #39~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar 17 21:16:15 UTC 2
server version: 0.7.46
plugin version: @scrypted/ring 0.0.109
########################
08/04/2023, 03:03:52
########################
node modules /home/innes/.scrypted/volume/plugins/@scrypted/ring/node18-linux-x64
native dependencies (outdated) got socket.io-client
Removing old dependencies: /home/innes/.scrypted/volume/plugins/@scrypted/ring/node18-linux-x64-20230329
plugin start/fork failed Error: EACCES: permission denied, mkdir '/home/innes/.scrypted/volume/plugins/@scrypted/ring/node18-linux-x64'
    at Object.mkdirSync (node:fs:1396:3)
    at mkdirpNativeSync (/home/innes/.scrypted/node_modules/mkdirp/lib/mkdirp-native.js:29:10)
    at Function.mkdirpSync [as sync] (/home/innes/.scrypted/node_modules/mkdirp/index.js:21:7)
    at installOptionalDependencies (/home/innes/.scrypted/node_modules/@scrypted/server/dist/plugin/plugin-npm-dependencies.js:55:26)
    at Object.onLoadZip (/home/innes/.scrypted/node_modules/@scrypted/server/dist/plugin/plugin-remote-worker.js:218:77)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.loadZip (/home/innes/.scrypted/node_modules/@scrypted/server/dist/plugin/plugin-remote.js:584:28)
    at async RpcPeer.handleMessageInternal (/home/innes/.scrypted/node_modules/@scrypted/server/dist/rpc.js:583:37) {
  errno: -13,
  syscall: 'mkdir',
  code: 'EACCES',
  path: '/home/innes/.scrypted/volume/plugins/@scrypted/ring/node18-linux-x64'
}
